src/main/java/user/database/UserDataBaseGateway.java

```diff
@@ -0,0 +1,9 @@
+package user.database;
+
+public interface UserDataBaseGateway {
```

Comment:
seems to me this gateway does two things: checks for user login and check for user registeration

This seems to violate the interface segregation principle (consider having seperated interface UserDBRegGateway and UserDBLoginGateway?)

If there's good reason to not make such change, could you (may I?) add a method in this gateway that returns a list of users (i.e. the "userMap" in your UserFileGateway) in db.

Reply:
Thank you for your suggestions. 
I have refactored UserDBlogGateway Out of UserDBRegGateway. Now all gateways are responsible for one functionality, Not violating ISP anymore.
And I also added The gateway called "UserInteractionDBGateway" to retrieve User data as an Array of User.

Reply:
For the class dependency on entities, sadly, might be unavoidable for serialization...

Reply:
Np! Thx for adding the user interaction gateways!

Yea I couldn’t think of a way to avoid that dependency either….in the example serialization repo they made a _userlist_ class that has an attribute _list of users_, and most of their DB classes directly work with _userlist_ obj instead of _user_. (I don’t get why this solves the dependency problem as userlist is also considered an entity)